### PR TITLE
US Reverse Geocoding API added to SDK

### PR DESCRIPTION
**Pull request by DuncanBeutler**

### CODE REVIEW

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private Address address;
+
+    @Key("coordinate")
+    private Coordinate coordinate;
+
+    @Key("distance")
+    private double distance;
+
+    @Key("latitude")
+    private double latitude;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:18]: Remove this field from the Result class. It should only be provided on the Coordinate class

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Key("longitude")
+    private double longitude;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:21]: Remove this field from the Result class. It should only be provided on the Coordinate class

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Key("street")
+    private String street;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:24]: Remove this field from the Result class. It should only be provided on the Address class

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Key("city")
+    private String city;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:27]: Remove this field from the Result class. It should only be provided on the Address class

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Key("state_abbreviation")
+    private String stateAbbreviation;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:30]: Remove this field from the Result class. It should only be provided on the Address class

File: src/main/java/com/smartystreets/api/us_reverse_geo/Result.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Key("zipcode")
+    private String zipcode;
```
Comment (mdwhatcott) [inline src/main/java/com/smartystreets/api/us_reverse_geo/Result.java:33]: Remove this field from the Result class. It should only be provided on the Address class

File: src/main/java/examples/UsReverseGeoExample.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            ex.printStackTrace();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+        Result[] results = lookup.getResponse().getResults();
+
+        System.out.printf("Results for input: (%f, %f)\n\n", lookup.getLatitude(), lookup.getLongitude());
+        for (Result result : results) {
+            System.out.println("Latitude: " + result.getLatitude().toString());
```
Comment (mdwhatcott) [inline src/main/java/examples/UsReverseGeoExample.java:45]: Use `result.getCoordinate().getLatitude()`, etc...

File: src/test/java/com/smartystreets/api/us_reverse_geo/ClientTest.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    //region [ Single Lookup ]
+
+    @Test
+    public void testAddressLookupSerializedAndSentWithContext() throws Exception {
+        RequestCapturingSender capturingSender = new RequestCapturingSender();
+        URLPrefixSender sender = new URLPrefixSender("http://localhost/lookup", capturingSender);
+        FakeSerializer serializer = new FakeSerializer(null);
+        Client client = new Client(sender, serializer);
+
+        Lookup lookup = new Lookup(44.444444444, -111.111111111);
```
Comment (mdwhatcott) [inline src/test/java/com/smartystreets/api/us_reverse_geo/ClientTest.java:26]: I propose one of these values be something like `44.888888888` so we see it properly round up in the assertion.

**Review by mdwhatcott — CHANGES_REQUESTED** [review]
The main thing is to NOT publish any fields that are deprecated (such as coordinate and address fields at the top-level).

**Review by mdwhatcott — APPROVED** [review]

**Comment by smccormickescreenlogic** [comment]
What version of the SDK will this be available in? I don't see it in 3.6.1. Or am I doing something wrong?

**Comment by DuncanBeutler** [comment]
It will be on 3.7.0. It will be along shortly, thank you for asking - we will keep you updated.

**Comment by DuncanBeutler** [comment]
3.7.0 is available

**Comment by smccormickescreenlogic** [comment]
Thanks! Saves me switching to an HTTP call.
________________________________

**Pull request merged**